packages/wallets/src/components/WalletImport.tsx

```diff
@@ -43,11 +56,58 @@ export default function WalletImport() {
     },
   });
 
-  const { fields } = useFieldArray({
+  const { fields, replace } = useFieldArray({
     control: methods.control,
     name: 'mnemonic',
   });
 
+  const submitMnemonicPaste = (mnemonicList: string) => {
+    closeMnemonicPaste();
+    let mList = mnemonicList.match(/\b(\w+)\b/g);
+
+    if (mList == null) {
+      openDialog(
+        <AlertDialog>
+          <Trans>
+            Your pasted list does not include 24 valid mnemonic words.
+          </Trans>
+        </AlertDialog>
+      );
+      return;
+    }
+
+    const intersection = mList.filter(element => options.includes(element));
+
+    if (intersection.length != 24) {
+      openDialog(
+        <AlertDialog>
+          <Trans>
+            Your pasted list does not include 24 valid mnemonic words.
+          </Trans>
+        </AlertDialog>
+      );
+    } else {
+      const mnemonic = intersection.map((word) => ({ word }));
+      replace(mnemonic);
+    }
+  };
```

Comment:
Just a suggestion.
The lines around `openDialog` seems reduntant and you can reduce the loc.
How about
```
const intersection = mList?.filter(element => options.includes(element));
if(!intersection || intersection.length !== 24){
  openDialog(...);
  return;
}
const mnemonic = intersectino.map((word) => ({ word }));
replace(mnemonic);
```